src/pool/srv_target.c

```diff
@@ -570,6 +570,21 @@ ds_pool_tgt_ec_eph_query_abort(struct ds_pool *pool)
 	pool->sp_ec_ephs_req = NULL;
 }
 
+static void
+pool_fetch_hdls_ult_abort(struct ds_pool *pool)
+{
+	if (!pool->sp_fetch_hdls)
+		return;
+
+	ABT_mutex_lock(pool->sp_mutex);
+	ABT_cond_signal(pool->sp_fetch_hdls_cond);
+	ABT_mutex_unlock(pool->sp_mutex);
+
+	ABT_mutex_lock(pool->sp_mutex);
```

Comment:
It is not your patch issue, but I do not understand the purpose of unlock and lock again, it does not guarantee that others can obtain the mutex during such very short interval.

Reply:
That's required by ABT API, I suppose it's for avoiding race of "adding to wait queue" and "wakeup".

Anyway, I hope we could replace all these ad-hoc ULT abort functions with sched_req_wait() call.